app/models/allocation.rb

```diff
@@ -2,4 +2,37 @@ class Allocation < ActiveRecord::Base
   belongs_to :country
   belongs_to :protected_area
   belongs_to :user
+
+  def self.to_csv
+    csv = ''
+    allocation_columns = Allocation.new.attributes.keys
+    allocation_columns.delete("created_at")
+    allocation_columns.delete("updated_at")
+    allocation_columns.delete("country_id")
```

Comment:
Since the columns you need are just a few, you could create an array where you manually put the name for those rather then deleting not needed columns.
Otherwise, if you want to keep deleting them, you could use this shorter syntax instead:
`array_with_needed_columns - array_with_unwanted_columns`

Reply:
Thanks @Levia I will fix this.